app/models/viewmodels/NotificationPageSingleElementFactory.scala

```diff
@@ -84,14 +83,30 @@ class NotificationPageSingleElementFactory @Inject()(decoder: Decoder) {
     notification: NotificationFrontendModel
   )(implicit messages: Messages): NotificationsPageSingleElement = {
 
-    val content = notification.actionCode.map { code =>
-      s"<p>${messages(decoder.actionCode(code).map(_.contentKey).getOrElse(""))}</p>"
-    }
+    val actionCodeExplanation = for {
+      code <- notification.actionCode
+      actionCode <- decoder.actionCode(code)
+      content = s"<p>${messages(actionCode.contentKey)}</p>"
+    } yield content
+
+    val errorExplanation = (for {
+      code <- notification.errorCodes
+      errorCode <- decoder.errorCode(code) match {
+        case None =>
+          logger.warn(s"Received inventoryLinkingControlResponse with unknown error code: $code")
+          None
+        case knownCode => knownCode
+      }
```

Comment:
This code return the same element. What do you think about something like this

```
def method(a: Option[String]): Option[String] =
  if(a.isEmpty) logger.warn(...)
  a
}
```

Of course with better naming or without function

```
errorCode <- {
  val code = decoder.errorCode(code)
  if(code.isEmpty) logger.warn(...)
  code
}
```

Reply:
:+1:

Reply:
Used you second solution.